src/platforms/javascript/common/session-replay/understanding-sessions.mdx

```diff
@@ -93,3 +93,75 @@ await replay.flush();
 ```
 
 In `session` mode, this will upload any pending recording data to Sentry. In `buffer` mode, this will upload any pending recording data to Sentry and then continue recording, the same as when an error is sampled with `replaysOnErrorSampleRate`.
+
+## Examples of Custom Sampling
+
+### Replays For Specific Users
+
+Here is an example of always recording a replay for a specific segment of your users. In this case, we would like to have error-based sampling for all users and we want to enable session-based replays for employees only. The `user` object has a field `isEmployee` to express if they are an employee or not.
+
+```javascript
+const replay = new Sentry.Replay();
+
+/**
+ * Initialize the Sentry SDK as normal.
+ * `replaysSessionSampleRate` is set to "0" so that we have manual control over session-based replays
+ * `replaysOnErrorSampleRate` is set to "1.0" to always record a replay when an error occurs
+ */
+Sentry.init({
+  dsn: "...",
+  replaysSessionSampleRate: 0,
+  replaysOnErrorSampleRate: 1.0,
+  integrations: [replay],
+});
+
+// ...
+// Sometime after your users are authenticated, check if they are an employee
+// If they are, we call `replay.start()` to start recording a replay for the employee
+// ...
+
+// Check if user is an employee, if so, always record a replay
+if (loggedInUser.isEmployee) {
+  // You can get a reference to the Sentry Replay integration like so:
+  const replay = Sentry.getCurrentHub().getIntegration(Sentry.Replay);
+  // or alternatively
+  // const replay = Sentry.getCurrentHub().getClient().getIntegrationById('Replay');
+
+  // Start the replay!
+  replay.start();
```

Comment:
This will not actually work! As since an error sample rate is set, replay is enabled and we'll run into this branch in `start()`:

```js
 if (this._isEnabled && this.recordingMode === 'buffer') {
      throw new Error('Replay buffering is in progress, call `flush()` to save the replay');
    }
```

I think you'd either need to do this:

```js
await replay.stop();
replay.start();
```

_or_ do this:

```js
replay.flush(); // <-- convert the error session to a regular session and start recording it
```